sklearn/utils/estimator_checks.py

```diff
@@ -923,6 +925,28 @@ def check_estimator_sparse_array(name, estimator_orig):
         _check_estimator_sparse_container(name, estimator_orig, sparse.csr_array)
 
 
+def check_f_contiguous_array_estimator(name, estimator_orig):
+    # Non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/23988
+    # https://github.com/scikit-learn/scikit-learn/issues/24013
+    estimator = clone(estimator_orig)
+
+    rng = np.random.RandomState(0)
+    X = 3 * rng.uniform(size=(20, 3))
+    X = _enforce_estimator_tags_X(estimator_orig, X)
```

Comment:
Why not use the same code as before (e.g. make_blobs)?

Reply:
Cause not all estimators pass with that. I got some "this solver can't ..." kind of error with that dataset. This `X, y` is what's used in many of the other tests in this file. I intend to clean that up as well, and try to avoid creating datasets inside tests.